src/Factories/Transform/UserSchema/OptionalIfTransformFactory.cs

```diff
@@ -0,0 +1,36 @@
+using System.Linq;
+using System.Threading.Tasks;
+using form_builder.Models;
+
+namespace form_builder.Factories.Transform.UserSchema
+{
+    public class OptionalIfTransformFactory : IUserPageTransformFactory
+    {
+        public async Task<Page> Transform(Page page, FormAnswers convertedAnswers)
+        {
+            var optionalIfElements = page.Elements
+                .Where(element => !string.IsNullOrEmpty(element.Properties.OptionalIfValue) 
+                || !string.IsNullOrEmpty(element.Properties.OptionalIfNotValue));
+
+            if (optionalIfElements.Any())
+            {
+                foreach (var element in optionalIfElements)
+                {
+                    var userChoice = convertedAnswers.AllAnswers
```

Comment:
I still like to see the Type if it's a "complicated" linq statement... and not sure it's always going to a "user choice"... it's just a question Id that has a value... go with more generic but hopefully still readable names.